go/tools/builders/link.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// link combines the results of a compile step using "go tool link". It is invoked by the
+// Go rules as an action.
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"go/build"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+func run(args []string) error {
+	// process the args
+	if len(args) < 2 {
+		return fmt.Errorf("Usage: link gotool [link args] -- <go link args>")
+	}
+	gotool := args[0]
+	args = args[1:]
+	linkargs := []string{}
+	goopts := []string{}
+	bctx := build.Default
+	bctx.CgoEnabled = true
+	for i, s := range args {
+		if s == "--" {
+			if i < len(args) {
```

Comment:
yes, you are right, I was not reading my own code correctly (originally there was a thing where the first value was not stripped, and the iteration index did not match the slice index, and I forgot I stopped doing that because it was messy)